src/som/compiler/MethodBuilder.java

```diff
@@ -521,30 +511,14 @@ public ReturnNonLocalNode getNonLocalReturn(final ExpressionNode expr,
         getOuterSelfContextLevel(), source);
   }
 
-  public MethodBuilder getOuterBuilder() {
-    return outerBuilder;
-  }
-
-  public MixinBuilder getEnclosingMixinBuilder() {
-    if (this.directOuterMixin == null) {
-      if (outerBuilder == null) {
-        return null;
-      } else {
-        return outerBuilder.getEnclosingMixinBuilder();
-      }
-    } else {
-      return directOuterMixin;
-    }
-  }
-
   public ExpressionNode getOuterRead(final String outerName,
       final SourceSection source) throws MixinDefinitionError {
-    MixinBuilder enclosing = getEnclosingMixinBuilder();
+    MixinBuilder enclosing = getOuterSelf();
```

Comment:
So, the whole notion of 'enclosing' was eliminated?
Not sure yet whether that's a good or bad thing.

Could you do a fuzzy search over the code to make sure we are consistent?
And, I want good definitions of `context` and `outer`, and all relevant methods should have a comment, to reflect the definitions. Pretty please :)
I know, I am asking more from you than from myself. But, yeah, since we now have the opportunity to do it right, we might be nicer to our future selfs then I was...